lxd/identities.go

```diff
@@ -166,6 +174,358 @@ func identityAccessHandler(authenticationMethod string, entitlement auth.Entitle
 	}
 }
 
+// swagger:operation POST /1.0/auth/identities/tls?public identities identities_post_tls_untrusted
+//
+//  Add a TLS identity
+//
+//  Adds a TLS identity as a trusted client.
+//  In this mode, the `token` property must be set to the correct value.
+//  The certificate that the client sent during the TLS handshake will be added.
+//  The `certificate` field must be omitted.
+//
+//  The `?public` part of the URL isn't required, it's simply used to
+//  separate the two behaviors of this endpoint.
+//
+//  ---
+//  consumes:
+//    - application/json
+//  produces:
+//    - application/json
+//  parameters:
+//    - in: body
+//      name: TLS identity
+//      description: TLS Identity
+//      required: true
+//      schema:
+//        $ref: "#/definitions/IdentitiesPostTLS"
+//  responses:
+//    "201":
+//      $ref: "#/responses/EmptySyncResponse"
+//    "400":
+//      $ref: "#/responses/BadRequest"
+//    "403":
+//      $ref: "#/responses/Forbidden"
+//    "500":
+//      $ref: "#/responses/InternalServerError"
+
+// swagger:operation POST /1.0/auth/identities/tls identities identities_post_tls
+//
+//	Add a TLS identity.
+//
+//	Adds a TLS identity as a trusted client, or creates a pending TLS identity and returns a token
+//	for use by an untrusted client. One of `token` or `certificate` must be set.
+//
+//	---
+//	consumes:
+//	  - application/json
+//	produces:
+//	  - application/json
+//	parameters:
+//	  - in: body
+//	    name: TLS identity
+//	    description: TLS Identity
+//	    required: true
+//	    schema:
+//	      $ref: "#/definitions/IdentitiesPostTLS"
+//	responses:
+//	  "201":
+//	    oneOf:
+//	      - $ref: "#/responses/CertificateAddToken"
+//	      - $ref: "#/responses/EmptySyncResponse"
+//	  "400":
+//	    $ref: "#/responses/BadRequest"
+//	  "403":
+//	    $ref: "#/responses/Forbidden"
+//	  "500":
+//	    $ref: "#/responses/InternalServerError"
+func createTLSIdentity(d *Daemon, r *http.Request) response.Response {
+	s := d.State()
+
+	// Parse the request.
+	req := api.IdentitiesTLSPost{}
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	localHTTPSAddress := s.LocalConfig.HTTPSAddress()
+
+	trusted, err := request.GetCtxValue[bool](r.Context(), request.CtxTrusted)
+	if err != nil {
+		return response.SmartError(fmt.Errorf("Failed to get authentication status: %w", err))
+	}
+
+	var cert *x509.Certificate
+	if trusted {
```

Comment:
In the interests of starting to reduce our cyclomatic complexity:

https://ticsdocumentation.tiobe.com/latest/docs/metrics/index.html

I think it would be worth separating this logic into a separate function called something like `createIdentityTLSValidate()`?

Reply:
Yep that's fine, the handler is getting a bit long.